src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -598,23 +607,37 @@ bool isSupportedConversion(Conversion conversion, BoundExpression operandOpt)
                     case ConversionKind.Identity:
                     case ConversionKind.DefaultOrNullLiteral:
                     case ConversionKind.ImplicitReference:
+                    case ConversionKind.ExplicitReference:
                         return true;
                     case ConversionKind.ImplicitTupleLiteral:
+                    case ConversionKind.ExplicitTupleLiteral:
+                        switch (operandOpt.Kind)
```

Comment:
This concerns me a bit in combination with line 629. Is there a guarantee that `arguments[i]` is a `BoundConversion` if `conversions[i]` is a tuple literal conversion?